src/mock-assembly/MockAssembly.cs

```diff
@@ -315,4 +321,43 @@ public class GrandChildClass
             }
         }
     }
+
+    [TestFixture]
+    public class FixtureWithAttachment
+    {
+        public const int Tests = 2;
+
+        public static readonly string Attachment1Name = "attachment1.txt";
+        public static readonly string Attachment1Contents = "CONTENTS";
+        public static readonly string Attachment1Description = "A description with some <values>";
+
+        public static readonly string Attachment2Name = "attachment2.txt";
+        public static readonly string Attachment2Contents = "CONTENTS2";
+        public static readonly string Attachment2Description = "second description";
+
+        [Test]
+        public void SingleAttachmentTest()
+        {
+            var context = TestContext.CurrentContext;
+            var filepath = Path.Combine(context.WorkDirectory, Attachment1Name);
+
+            File.WriteAllText(filepath, Attachment1Contents);
+            TestContext.AddTestAttachment(filepath, Attachment1Description);
+        }
+
+        [Test]
+        public void MultiAttachmentTest()
+        {
+            var context = TestContext.CurrentContext;
+            var filepath1 = Path.Combine(context.WorkDirectory, Attachment1Name);
+            var filepath2 = Path.Combine(context.WorkDirectory, Attachment2Name);
+
+            File.WriteAllText(filepath1, Attachment1Contents);
+            TestContext.AddTestAttachment(filepath1, Attachment1Description);
+
+            File.WriteAllText(filepath2, Attachment2Contents);
+            TestContext.AddTestAttachment(filepath2, Attachment2Description);
```

Comment:
Please consider the xml to be final (or at least, to be something we would make backwards compatible!). That part was released as part of NUnit Framework 3.7 - the remaining issues are only still open as I've been a little tied up. Now you've sorted this one, and it's just the docs left, I'll try and do it soon. :-)

I'll let Charlie comment on the tests.